scanner/src/funTest/kotlin/HttpCacheTest.kt

```diff
@@ -73,41 +82,165 @@ class HttpCacheTest : StringSpec() {
-                    vcs = VcsInfo.EMPTY
-            )
-
-            val resultFile = createTempFile()
-            val resultContent = "magic"
-
-            // Put the file contents into the cache.
-            resultFile.writeText(resultContent)
-            cache.write(pkg, resultFile) shouldBe true
-
-            // Delete the original result file to ensure it gets re-created.
-            resultFile.delete() shouldBe true
-
-            // Get the file contents from the cache.
-            cache.read(pkg, resultFile) shouldBe true
-            resultFile.readText() shouldEqual resultContent
-
-            // Clean up.
-            resultFile.delete() shouldBe true
+            val result = cache.add(id, scanResult)
+            val cachedResults = cache.read(id)
+
+            result shouldBe false
+            cachedResults.id shouldBe id
+            cachedResults.results.size shouldBe 0
         }
+
+        "Can retrieve all scan results from cache" {
+            val cache = ArtifactoryCache("http://${loopback.hostAddress}:$port", "apiToken")
+            val scanResult1 = ScanResult(provenanceWithSourceArtifact, scannerSpecs1, scanSummaryWithFiles,
+                    rawResultWithContent)
+            val scanResult2 = ScanResult(provenanceWithSourceArtifact, scannerSpecs2, scanSummaryWithFiles,
+                    rawResultWithContent)
+
+            val result1 = cache.add(id, scanResult1)
+            val result2 = cache.add(id, scanResult2)
+            val cachedResults = cache.read(id)
+
+            result1 shouldBe true
+            result2 shouldBe true
+            cachedResults.results.size shouldBe 2
+            cachedResults.results should contain(scanResult1)
+            cachedResults.results should contain(scanResult2)
+        }
+
+        "Can retrieve all scan results for specific scanner from cache" {
+            val cache = ArtifactoryCache("http://${loopback.hostAddress}:$port", "apiToken")
+            val scanResult1 = ScanResult(provenanceWithSourceArtifact, scannerSpecs1, scanSummaryWithFiles,
+                    rawResultWithContent)
+            val scanResult2 = ScanResult(provenanceWithVcsInfo, scannerSpecs1, scanSummaryWithFiles,
+                    rawResultWithContent)
+            val scanResult3 = ScanResult(provenanceWithSourceArtifact, scannerSpecs2, scanSummaryWithFiles,
+                    rawResultWithContent)
+
+            val result1 = cache.add(id, scanResult1)
+            val result2 = cache.add(id, scanResult2)
+            val result3 = cache.add(id, scanResult3)
+            val cachedResults = cache.read(id, scannerSpecs1)
+
+            result1 shouldBe true
+            result2 shouldBe true
+            result3 shouldBe true
+            cachedResults.results.size shouldBe 2
+            cachedResults.results should contain(scanResult1)
+            cachedResults.results should contain(scanResult2)
+        }
+
+//        "HTTP GET returns what was PUT" {
```

Comment:
Reminder: This out-commented code should be removed.

Reply:
Done.